File: navigation/src/main/java/com/example/navigation/stepsEngine/field/CardTypeField.java
package com.example.navigation.stepsEngine.field;

import com.example.navigation.stepsEngine.enums.Type;
import com.example.navigation.stepsEngine.payment.FlowState;

public class CardTypeField extends Field {

    @Override
    public String get(FlowState flowState) {
        final Type type = flowState.getCard().getType();
        if (type == null) {
            return null;
        } else {
            return type.name();
        }
    }

    @Override
    public String getId() {
        return null;
    }

    @Override
    public void set(FlowState flowState, Object value) {

    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/field/CardTypeField.kt
package com.example.navigation.stepsEngine.field

import com.example.navigation.stepsEngine.enums.CardType
import com.example.navigation.stepsEngine.payment.FlowState

class CardTypeField : Field() {
    override operator fun get(flowState: FlowState): String? {
        val type = flowState.card.cardType
        return type?.name
    }

    override fun getId(): String {
        return FieldName.CARD_TYPE
    }

    override operator fun set(flowState: FlowState, value: Any?) {
        flowState.card.cardType = value as CardType
    }
}